### PR TITLE
Add JS versions of all docs code snippets

### PR DESCRIPTION
**Pull request by aomarks**
For this change I didn't want to modify any of the existing TS code or prose, because the language switch is still behind the `?mods=jsSamples` feature flag, so there will be a second pass with a few more changes that we'll merge as part of the public release.

Part of #332

### CODE REVIEW

File: packages/lit-dev-content/site/docs/composition/controllers.md [hunk 1]
```diff
@@ -70,8 +72,23 @@ class ClockController implements ReactiveController {
 }
 ```
 
+```js
+class ClockController {
+  constructor(host) {
+    // Store a reference to the host
+    this.host = host;
```
Comment (justinfagnani) [inline packages/lit-dev-content/site/docs/composition/controllers.md:79]: I wonder if we want to use class fields in JS examples? They're supported in all current browsers now.

**Review by justinfagnani — APPROVED** [review]

File: packages/lit-dev-content/site/docs/components/shadow-dom.md [hunk 2]
```diff
@@ -84,7 +84,7 @@ This decorator is equivalent to:
 
 ```js
 get _first() {
-  return this.renderRoot.querySelector('#first');
+  return this.renderRoot?.querySelector('#first');
```
[the remark this answers is not in the document]
Comment (aomarks) [reply packages/lit-dev-content/site/docs/components/shadow-dom.md:87]: Done.

File: packages/lit-dev-content/site/docs/composition/controllers.md (hunk 1 above)
Comment (aomarks) [reply packages/lit-dev-content/site/docs/composition/controllers.md:79]: In this case do you mean?:

```js
class ClockController {
  host;
  constructor(host) {
    // Store a reference to the host
    this.host = host;
    // Register for lifecycle updates
    host.addController(this);
  }
}
```

Comment (justinfagnani) [reply packages/lit-dev-content/site/docs/composition/controllers.md:79]: Yeah. Though I'm not even sure myself whether that's better or not...

Comment (aomarks) [reply packages/lit-dev-content/site/docs/composition/controllers.md:79]: I think I'll leave it off for now, but can revisit.

Comment (edimitchel) [reply packages/lit-dev-content/site/docs/composition/controllers.md:79]: Or may be more simple: 
```js
class ClockController {
  constructor(private host) {
    // Register for lifecycle updates
    host.addController(this);
  }
}
```

**Pull request merged**